yaml_edit/lib/src/editor.dart

```diff
@@ -266,7 +264,7 @@ class YamlEditor {
 
     if (parentNode is YamlList) {
       final expected = wrapAsYamlNode(
-        [...parentNode.nodes]..[keyOrIndex] = valueNode,
+        [...parentNode.nodes]..[keyOrIndex as int] = valueNode,
```

Comment:
What if this cast fails?

We probably have to do `isValidIndex(keyOrIndex, parentNode.length)`, and if this fails, then we have to do `_pathErrorOrElse` the same way `_traverse` would do..

You actually found a bug here, I think...

```yaml
['a','b']
```

If I was to do `editor.update([3], 'c')` it'll probably fail with some unfortunate error, rather than a `PathError` (which is extends ArgumentError).

Same if I were to do: `editor.update(["not-a-number"], 'c')`

Reply:
The index out of range is handled by `updateInList()` which throws a `RangeError`. I tried using `isValidIndex()` to check the index:

```dart
!isValidIndex(keyOrIndex, parentNode.length)
```
... but that failed tests that specifically expect a `RangeError` for outside ranges. For now I plan to use the following but let me know if you want to consolidate this so it will conform to how the other methods that use `_traverse()` handle this. 

```dart
    if (parentNode is YamlList) {
      if (keyOrIndex is! int) {
        throw PathError.unexpected(path,
            'Attempted to use "$keyOrIndex" as index on the list $parentNode');
      }
      final expected = wrapAsYamlNode(
        [...parentNode.nodes]..[keyOrIndex] = valueNode,
      );

      return _performEdit(updateInList(this, parentNode, keyOrIndex, valueNode),
          collectionPath, expected);
    }
```